ocean_lib/models/test/test_datatoken_order_both_templates.py

```diff
@@ -220,19 +221,24 @@ def test_buy_DT_and_order(
     publish_bal1 = USDC.balanceOf(consumer_wallet.address)
     provider_fee_bal1 = USDC.balanceOf(another_consumer_wallet.address)
 
-    tx = DT.buy_DT_and_order(
-        consumer=another_consumer_wallet.address,
-        service_index=1,
-        provider_fees=provider_fees,
-        consume_market_order_fee_address=consume_fee_address,
-        consume_market_order_fee_token=DAI.address,
-        consume_market_order_fee_amount=0,
-        exchange=exchange,
-        max_base_token_amount=to_wei(2.5),
-        consume_market_swap_fee_amount=to_wei(0.001),  # 1e15 => 0.1%
-        consume_market_swap_fee_address=another_consumer_wallet.address,
-        transaction_parameters={"from": publisher_wallet},
-    )
+    args = {
+        "consumer": another_consumer_wallet.address,
+        "service_index": 1,
+        "provider_fees": provider_fees,
+        "consume_market_fees": FeeTokenInfo(
+            address=consume_fee_address,
+            token=DAI.address,
+        ),
+        "exchange": exchange,
+        "transaction_parameters": {"from": publisher_wallet},
+    }
+
+    if template_index == 2:
+        args["max_base_token_amount"] = to_wei(2.5)
+        args["consume_market_swap_fee_amount"] = to_wei(0.001)  # 1e15 => 0.1%
+        args["consume_market_swap_fee_address"] = another_consumer_wallet.address
+
+    tx = DT.buy_DT_and_order(**args)
```

Comment:
Passing in `**args` to `buy_DT_and_order()` is harder to understand than before. Especially when `args` is still so big.

I recommend to have two unit tests for `buy_DT_and_order()`: one with good defaults and one for non-defaults. 

Here's a good-defaults call:

```python
order_tx_id = exchange.buy_DT_and_order({"from": publisher_wallet})
```

Notes:
- calls 'exchange` (`Exchange` class), not `DT` (`Datatoken` class). Just like `buy_DT()`. And `exchange` knows which datatoken to use, by definition
- it returns `order_tx_id`, not `tx`. Like calls to `pay_for_access_service()`.

Here's how the defaults would be set:
```python
consumer = transaction_parameters["from"].address
service_index = 1
provider_fees = (default to zero)
consume_market_fees = (default to zero)
```

And, here's a non-defaults call:
```python
provider_fees  = get_mock_provider_fees("MockDAI", publisher_wallet, valid_until=valid_until)
consume_market_fees": FeeTokenInfo(address=consume_fee_address, token=DAI.address)
order_tx_id = exchange.buy_DT_and_order(
  provider_fees=provider_fees,
  consume_market_fees=consume_market_fees,
  service_index=1,
  tx_dict={"from": publisher_wallet}
)
```